src/PlayerbotAI.cpp

```diff
@@ -4144,35 +4142,38 @@ bool PlayerbotAI::AllowActive(ActivityType activityType)
     if (sPlayerbotAIConfig->botActiveAlone <= 0)
         return false;
 
-    if (sPlayerbotAIConfig->botActiveAlone >= 100)
-        return true;
-
-    if (maxDiff > 1000)
-        return false;
-
-    uint32 mod = 100;
-
-    // if has real players - slow down continents without player
-    if (maxDiff > 100)
-        mod = 50;
-
-    if (maxDiff > 150)
-        mod = 25;
-
-    if (maxDiff > 200)
-        mod = 10;
-
-    if (maxDiff > 250)
+    // use the configured 'botActiveAlone 'percentages. Downscale the activity, within 
+    // reason ,with the intension to give the server some breathing space when needed.
+    uint32 mod = sPlayerbotAIConfig->botActiveAlone > 100 ? 100 : sPlayerbotAIConfig->botActiveAlone;
+    if (sPlayerbotAIConfig->botActiveAloneAutoScale) 
     {
-        if (Map* map = bot->GetMap())
-        {
-            if (map->GetEntry()->IsWorldMap())
+        uint32 maxDiff = sWorldUpdateTime.GetAverageUpdateTime();
+        if (maxDiff > 500)
+            return false;
+        if (maxDiff > 80)
+            mod = (mod * 9) / 10;
+        if (maxDiff > 100)
+            mod = (mod * 8) / 10;
+        if (maxDiff > 120)
+            mod = (mod * 7) / 10;
+        if (maxDiff > 140)
+            mod = (mod * 6) / 10;
+        if (maxDiff > 160)
+            mod = (mod * 5) / 10;
+        if (maxDiff > 180)
```

Comment:
`mod` here will be reduced repeatedly, is this intentional?

It seems that it should be judged from large to small, for example:
```
if (maxDiff > 180)
    mod = (mod * 3) / 10;
else if (maxDiff > 160)
    mod = (mod * 5) / 10;
else if
    ...
```

Reply:
/pacepalm, no it wasnt. Going back and forward in changes, made a mistake along the way. Good catch!